### PR TITLE
rcal-850 Addition of catalog step to mosaic pipeline

### LINKED ISSUE #1256: Add source catalog step to mosaic pipeline
_Issue [RCAL-850](https://jira.stsci.edu/browse/RCAL-850) was created on JIRA by [David Davis](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ddavis):_

Add the source catalog step to the mosaic pipeline to generate the L3 catalog on the mosaiced products.

### PR DESCRIPTION
**Pull request by ddavis-stsci**
Resolves [RCAL-850](https://jira.stsci.edu/browse/RCAL-850)

Closes #1256 

This PR adds the source catalog step to run on the resampled image at the end of the mosaic pipeline to create the
_cat.asdf 
_segm.asdf
files.

**Checklist**
- [x] added entry in `CHANGES.rst` under the corresponding subsection
- [x] updated relevant tests
- [x] updated relevant documentation
- [x] updated relevant milestone(s)
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below. [How to run regression tests on a PR](https://github.com/spacetelescope/romancal/wiki/Running-Regression-Tests-Against-PR-Branches)
https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/813/

### CODE REVIEW

File: romancal/pipeline/mosaic_pipeline.py [hunk 1]
```diff
@@ -77,7 +80,8 @@ def process(self, input):
             # check to see if the product name contains a skycell name & if true get the skycell record
             product_name = input.asn_table["products"][0]["name"]
             try:
-                skycell_name = product_name.split("_")[3]
+                skycell_name = product_name.split("_")[1]
+                skycell_name = input.asn_table["target"]
```
Comment (schlafly) [inline romancal/pipeline/mosaic_pipeline.py:84]: At present the product_name-based skycell_name is immediately overwritten by the target-based one?  Probably just delete line 83 if we have updated asn files?

**Review by schlafly — COMMENTED** [review]
LGTM, I flagged one bit I was confused by.

File: romancal/pipeline/mosaic_pipeline.py (hunk 1 above)
Comment (ddavis-stsci) [reply romancal/pipeline/mosaic_pipeline.py:83]: And now you know why it is a draft.

**Review by schlafly — APPROVED** [review]
LGTM, thanks.

**Pull request merged**